lib/oli/authoring/course.ex

```diff
@@ -338,4 +340,95 @@ defmodule Oli.Authoring.Course do
       description: "New family from project creation"
     }
   end
+
+  def get_project_survey(project_id) do
+    ProjectResource
+    |> join(:inner, [pr], rev in Revision, on: pr.resource_id == rev.resource_id)
+    |> join(:left, [_, rev], rev2 in Revision,
```

Comment:
This is not the correct way to retrieve the specific Revision for a resource from the authoring perspective.  Instead, you need to `join` through the `PublishedResource` schema on the `publication_id` for the unpublished publication for this project.  Look at the functions in `Oli.Publishing.AuthoringResolver` for examples of how to do this.

Reply:
Thanks for this! I'm still wrapping my head around all the relations 😅